### ISSUE TITLE
Closing window does not close Application

### ISSUE THREAD

**Issue by wehowski** [jyapayne/Web2Executable#152]
If I have one window left (without menu) and click the red X in the upper right corner, the process of the application is still present in the taskmanager, preventing to restart a new instance of the app.

**Comment by jyapayne** [comment]
Does your app open subprocesses while it is running?

**Comment by wehowski** [comment]
Yes and no.
So far, it SHOULD only start subprocesses in the update/upgrade process.
I started a workaround but have to compile and tests this first before I can give report.
So fa this is my background process: http://cdn.webfan.de/cdn/application/webfan/node_modules/webfan/navigator/-webkit-webfan.js?minify=no

````
   Used.gui.Window.get().on('close', function() {
    // Hide the window to give user the feeling of closing immediately
     this.hide();

     // If the new window is still open then close it.
     //if (win != null)
    //  win.close(true);
    try{
		(window.top || window).close();
	}catch(err){
		
	}

    
      if(0 === chrome.app.window.getAll().length){
	    process.nextTick(function(){	
	  	  navigator.webfan.quit();
	  	});  
	  }

    // After closing the new window, close the main window.
    this.close(true);
    

  });

````
At "Window Settings" I have option "Show" disabled and open windows like described in the other thread?

However, it apperead when a window is opened and I click the red X there are still proceesses active in the task manager.

Also, I tried this:
```
chrome.idle.setDetectionInterval(60);
chrome.idle.onStateChanged.addListener(function( newState ) {


 if('idle' === newState){		 
      setTimeout(function(){
        if( (true===navigator['-webkit-frdl'].shutdown 
           || 0 === chrome.app.window.getAll().length
         ) && 'idle' === newState){
 	          navigator.webfan.quit();
        }  	
      },5000); 
  }else if('active' === newState){
  	 /* todo ? */
  }else if('locked' === newState){
  	 /* todo ? */
  }
      
});
```

----
OT: I compile at my windows home PC, to compile for every system and then upload takes 2-3 hours for me right now, I think of to compile at my linux virtual server direct, but I don't trust me (have no experience about that)!
----

```
function quit(app){
var a = app;
 process.nextTick(function(){ 
        setTimeout(function(){
        	if('undefined' !== typeof frdl){
        		frdl.each(chrome.app.window.getAll(), function(i,Win){
        	       Win.close();	
            	});
            }else{
				console.warn('Exit Application without µ.Flow.Library loaded (frdl missing)');
			}
			 process.nextTick(function(){
                a.quit();
             });   
        },1);
 });
}
```

The above is the navigator.webfan.quit() function I call when clicking "exit" in the app, but when clicking the red x in the right window cornor, there are still process in the taskmanager and that prevents to restart the app again.

There is also
```
Used.gui.App.on('open', function(cmdline) {
  	  try{
  	        setTimeout(function(){
                    process.nextTick(function(){ 
                       setTimeout(function(){
                         	   console.log('(re.)open command line: ' + cmdline);
                       	   
                       },1);
                    }); 
             },1);         
         	
	  }catch(err){
	  	 if('undefined' !== typeof console)console.error(err);
	  }
           	     
});
```
but without any function so far.

**Comment by wehowski** [comment]
OK, I recompiled my app to v0.0.34-beta and it still appears that if click the red x in the upper right corner, the process in taskmanager is still active. 

No, the application should not have started any sub-process at this time!?

Maybe there is to mention: If I click the .exe there are up to 4 processes in the taskmanager (nwjs), one in the group "apps" and three in the group "backgroundprocesses".
This issue maybe is related or similar to this one: https://github.com/jyapayne/Web2Executable/issues/151

Although the code is not "fully open" it is not protected yet, so you can check it if you like: 
Compiled versions:  http://frdl.github.io/frdl/install/checksum/-webkit-webfan.json
Source lately:  http://cdn.webfan.de/cdn/application/webfan/*

**Comment by jyapayne** [comment]
This is indeed strange behaviour. I can't access your source code from http://cdn.webfan.de/cdn/application/webfan/* because it says "404 not found". Is there another way to access it? I also cannot access your compiled versions with the same error.

**Comment by wehowski** [comment]
Oh sorry, please just cut the filenames:
cdn.webfan.de/cdn/application/install/webfan-v0.0.34-beta/windows-x64/
/cdn/application/install/webfan-v0.0.34-beta/windows-x32/
/cdn/application/install/webfan-v0.0.34-beta/mac-x64/
/cdn/application/install/webfan-v0.0.34-beta/linux-x64/
/cdn/application/install/webfan-v0.0.34-beta/linux-x32/

I also temporary added a config.xml file wich makes all my source code a downloadable widget.zip at cdn.webfan.de/cdn/application/webfan/ 
This is not what I want, so please give me a reply if you downloaded the code so I can delete this config.xml/.widget.zip.

But all files are accesable then via /cdn/application/webfan/*FILENAME 
Sorry for the misunderstanding about the urls, my failure

**Comment by jyapayne** [comment]
@wehowski, I'm sorry, but it's really hard to follow your code. If there's any way you can come up with a minimal example, that would be extremely helpful. I simply cannot reproduce your behavior when I open my test applications. Also, this is more of an NW.js issue, since my code does not interact with anything after the application is exported. Please file a similar issue on the NW.js repo.
